### PR TITLE
Adding EdgeModifier support for chain()

### LINKED ISSUE #17083: Update chain() to support Labels
**Description**
The `airflow.models.baseoperator.chain()` is a very useful and convenient way to add sequential task dependencies in DAGs.  This function has [recently been updated](https://github.com/apache/airflow/issues/16635) to support `BaseOperator` and `XComArgs` but should also be able to support `Labels` as well.

**Use case / motivation**
Users who create tasks via the `@task` decorator will not be able to use the `chain()` function to apply sequential dependencies that do not share an `XComArg` implicit dependency with a `Label`.  This use case can occur when attempting to chain multiple branch labels and the next sequential task.

With the new update (yet to be released), users will receive the following exception when attempting to chain an `XComArg` and `Label`:

```bash
TypeError: Chain not supported between instances of <class 'airflow.utils.edgemodifier.EdgeModifier'> and <class 'airflow.models.xcom_arg.XComArg'>
```

**Are you willing to submit a PR?**
Absolutely. 🚀 

**Related Issues**
None

### PR DESCRIPTION
**Pull request by josh-fell**
Closes: #17083

- Added support for EdgeModifier in chain() helper function
- Added additional tests in test_baseoperator.py

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/models/baseoperator.py [hunk 1]
```diff
@@ -1546,13 +1546,17 @@ def inherits_from_dummy_operator(self):
         return getattr(self, '_is_dummy', False)
 
 
-def chain(*tasks: Union[BaseOperator, "XComArg", Sequence[Union[BaseOperator, "XComArg"]]]):
+def chain(
+    *tasks: Union[
+        BaseOperator, "XComArg", EdgeModifier, Sequence[Union[BaseOperator, "XComArg", EdgeModifier]]
```
Comment (uranusjr) [inline airflow/models/baseoperator.py]: At this point we should make this `BaseOperator, "XComArg", EdgeModifier` thing a type alias, maybe something like

```python
Chainable = Union[BaseOperator, "XComArg", EdgeModifier]

def chain(*tasks: Union[Chainable, Sequence[Chainable]]) -> None:
    ...
```

Comment (josh-fell) [reply airflow/models/baseoperator.py]: Brilliant. I'm sure there will be other types that will be "chainable" in the future.  I agree the annotations are beginning to become unwieldy.

**Review by uranusjr — APPROVED** [review]

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
Tests are failing @josh-fell

**Comment by josh-fell** [comment]
> Tests are failing @josh-fell

Some of these failures look to be resource-related. Rebasing and let's see how the new test runs turn out.

**Comment by josh-fell** [comment]
@kaxil Not as many failures after the rebase.

The 4 failing MSSQL tests are failing for the same reason in [this issue comment](https://github.com/apache/airflow/issues/17088#issuecomment-883032995).

The failing Postgres and MySQL jobs pass locally for the offending test "test_scheduler_job.py":
` ./breeze --python 3.9 --backend postgres --postgres-version 13 tests tests/jobs/test_scheduler_job.py`
image


` ./breeze --python 3.6 --backend mysql --mysql-version 5.7 tests tests/jobs/test_scheduler_job.py`
image

How should I proceed?

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Looks like transient errors only. Merging

**Pull request merged**